### PR TITLE
Swap from Go Dep to Go Modules.

### PR DESCRIPTION
**Pull request by nicfb**
This addresses issue #31

### CODE REVIEW

**Comment by nitishm** [comment]
Looks good to me visually. Need to try it out quickly before I merge it in. Thanks @nicfb 🎉

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module go-rejson
```
Comment (Shivam010) [inline go.mod]:
```suggestion
module github.com/nitishm/go-rejson
```
This way the indirect self-dependency of the project will be resolved.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+go 1.12
```
Comment (Shivam010) [inline go.mod]:
```suggestion
```
Also, I think that hard binding 'go' directive value of '1.12' should be removed

xref: golang/go#30446

Comment (nicfb) [reply go.mod]: > Also, I think that hard binding 'go' directive value of '1.12' should be removed
> 
> xref: [golang/go#30446](https://github.com/golang/go/issues/30446)

So this should be fine being omitted all together? @Shivam010

Comment (nitishm) [reply go.mod]: Yes you dont need to specify the go directive. We should remove that. Reminds me that the builds are failing on older version. Maybe I should've retained the Gopkg.* files 🤔 . WDYT @Shivam010 how do we support pre-go 1.11.x versions ?

**Comment by nitishm** [comment]
I think it's best to revert this PR. We need to consider v2.0.0 of the go-rejson module with go-modules and change all import paths as well. This is why you see the v2.0.0 incompatible statement in the go.mod file. My mistake for not looking closely. 
@nicfb let's try to do this more cleanly while fixing all import paths. Let's discuss this further on the original ticket.

File: go.mod (hunk 2 above)
Comment (Shivam010) [reply go.mod]: > Yes you dont need to specify the go directive. We should remove that. Reminds me that the builds are failing on older version. Maybe I should've retained the Gopkg.* files 🤔 . WDYT @Shivam010 how do we support pre-go 1.11.x versions ?

I think omitting 1.12 directive and using module path `github.com/nitishm/go-rejson` will not fail builds for the older versions, as go semantics will help.

**Pull request merged**